crates/core/tests/integration.rs

```diff
@@ -540,7 +541,7 @@ fn test_prune(
     let plan = repo.prune_plan(&prune_opts)?;
     // TODO: Snapshot-test the plan (currently doesn't impl Serialize)
     // assert_ron_snapshot!("prune", plan);
-    plan.do_prune(&repo, &prune_opts)?;
+    prune_repository(&repo, plan, &prune_opts)?;
```

Comment:
First, we could also implement a `plan.do_prune` on `PrunePlan` and mark it as deprecated. Then we could have this PR as non-breaking.

Second, I would not make a `pub prune_repository`, but instead make this a `Repository` method. Then it would be simply `repo.prune(plan, &prune_opts)` here...

Reply:
I agree in general, I did this, because it is relatively inconsistent over `rustic_core`, e.g. we have [pub(crate) fn copy<'a, Q, R: IndexedFull, P: ProgressBars, S: IndexedIds>(](https://github.com/rustic-rs/rustic_core/blob/f68ffa3fe1616b66b3f1b944c981155c30b86534/crates/core/src/commands/copy.rs#L44), [pub(crate) fn backup<P: ProgressBars, S: IndexedIds>(](https://github.com/rustic-rs/rustic_core/blob/f68ffa3fe1616b66b3f1b944c981155c30b86534/crates/core/src/commands/backup.rs#L214), [pub(crate) fn dump<P, S: IndexedFull>(](https://github.com/rustic-rs/rustic_core/blob/f68ffa3fe1616b66b3f1b944c981155c30b86534/crates/core/src/commands/dump.rs#L28), etc.

Reply:
So that was my intention, to make it more consistent. Each command has an entry point, and we don't implement everything on Repository, because then we get a huge master object with a lot of methods, which makes it also harder to test and couples all that functionality to that object/type. :/

Reply:
`repo.prune()` we can still implement for convenience using `prune_repository`?

Reply:
Implemented `prune()` on repo, which uses `prune_repository()`.